### DISCUSSION TITLE
Template panel

### DISCUSSION THREAD

**Discussion by dnaldoog** [RomanKubiak/ctrlr#509] [Show and tell]
This panel contains the basic structure code I always use when creating a new panel.
Feel free to use it as a starting template for your own work. 🎨

Version 1.0

https://sourceforge.net/projects/ctrlr-utilities/files/developer%20tools/template/

![template_image (1)](https://user-images.githubusercontent.com/76831683/206592059-2ff2a361-d071-485a-8659-4144b7029de1.png)

**Comment by dnaldoog** [comment]
Version 2.0 released - includes caching for _uiTabs_

https://sourceforge.net/projects/ctrlr-utilities/files/developer%20tools/template/

**Reply by unityconstruct** [reply]
BTW... salivated when I ran into your repo there... its a treasure trove

**Comment by unityconstruct** [comment]
nice! templates are the plagiarist's best fried

**Reply by Ted-juh** [reply]
Because the functions/ methods are built-in in ctrlr it's possible to bake-in what you made as a separate panel. Question is if everyone wants to maintain this workflow.

**Comment by dnaldoog** [comment]
Here's a new more basic template that I am using. The _uiSliders_ A-H are used for repainting a _uiCustomComponent_ when positioning elements therein.

### EDIT: Updated 6/5/2026

[TEMPLATE_3_0.zip](https://github.com/user-attachments/files/28618170/TEMPLATE_3_0.zip)
